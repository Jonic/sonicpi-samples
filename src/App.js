import React, { Component } from 'react'
import Player from './components/Player/Player'

class App extends Component {
  constructor() {
    super()

    this.state = {
      isLoading: false,
    }
  }

  updateIsLoading = (newState) => {
    this.setState({
      isLoading: newState,
    })
  }

  render() {
    return (
      <Player
        isLoading={this.state.isLoading}
        updateIsLoading={this.updateIsLoading}
      />
    )
  }
}

export default App
