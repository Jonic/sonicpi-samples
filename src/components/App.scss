@import '../styles/component-helper';

@keyframes bg-color-fade {
  0%, 100% { background-color: $color-primary-navy;      }
  12.5%    { background-color: $color-primary-blue;      }
  25%      { background-color: $color-primary-turquoise; }
  37.5%    { background-color: $color-primary-green;     }
  50%      { background-color: $color-primary-yellow;    }
  62.5%    { background-color: $color-primary-orange;    }
  75%      { background-color: $color-primary-red;       }
  87.5%    { background-color: $color-primary-violet;    }
}

$App-background-color:           $color-primary-navy;
$App-color:                      $color-white;
$App-transition-duration:        $transition-duration-extremely-long;
$App-transition-timing-function: $transition-timing-eased;
$App-padding:                    3vw;

.c-App {
  animation-duration:        $App-transition-duration;
  animation-fill-mode:       forwards;
  animation-iteration-count: infinite;
  animation-name:            bg-color-fade;
  animation-timing-function: $App-transition-timing-function;
  background-color:          $App-background-color;
  color:                     $App-color;
  padding:                   $App-padding;
  height:                    100vh;
  width:                     100vw;
}

.c-App__title {
  @include iota-type(10vw, 1.5);

  font-style:  italic;
  font-weight: $font-weight-light;
  text-align:  center;
}
