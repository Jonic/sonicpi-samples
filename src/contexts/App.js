import React, { useEffect, useState } from 'react'

import PropTypes from 'prop-types'
import { trackPageView } from '../helpers/Analytics'

const AppContext = React.createContext({
  isLoading: true,
  setIsLoading: () => {},
})

const AppContextProvider = ({ children }) => {
  const [isLoading, setIsLoading] = useState(true)

  useEffect(() => trackPageView(), [])

  const contextValue = {
    isLoading,
    setIsLoading,
  }

  return (
    <AppContext.Provider value={contextValue}>{children}</AppContext.Provider>
  )
}

AppContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
}

export default AppContext
export { AppContextProvider }
