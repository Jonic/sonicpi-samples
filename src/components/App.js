import './App.scss'

import AppContext, { AppContextProvider } from 'contexts/App'
import React, { useContext } from 'react'

import Player from './Player/Player'
import Visualizer from './Visualizer/Visualizer'
import classnames from 'classnames'

const App = () => {
  const { isLoading } = useContext(AppContext)
  const cAppClassNames = classnames(['c-app', isLoading && 'c-app--is-loading'])

  return (
    <AppContextProvider>
      <main className={cAppClassNames}>
        <h1 className="c-app__title">Sonic Pi Samples</h1>

        <section className="c-app__player">
          <Player />
        </section>
      </main>

      <Visualizer />
    </AppContextProvider>
  )
}

export default App
