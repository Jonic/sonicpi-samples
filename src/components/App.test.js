jest.mock('../helpers/Analytics')

import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

let wrapper

describe('<App />', () => {
  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  })

  it('renders without crashing', () => {
    expect(wrapper.find('.c-App')).toBePresent()
  })
})
