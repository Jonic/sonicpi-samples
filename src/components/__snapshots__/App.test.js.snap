// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders... matches its snapshot 1`] = `
<div
  className="c-App"
>
  <h1
    className="c-App__title"
  >
    Sonic Pi Samples
  </h1>
   
  <Player
    isLoading={false}
    toggleIsLoading={[Function]}
    updateIsLoading={[Function]}
  />
</div>
`;
